cluster/terraform_kubernetes/prometheus.tf

```diff
@@ -0,0 +1,164 @@
+resource "kubernetes_cluster_role" "prometheus" {
+  count = var.enable_monitoring ? 1 : 0
+
+  metadata {
+    name = "prometheus"
+  }
+
+  rule {
+    api_groups = [""]
+    resources  = ["nodes", "nodes/proxy", "services", "endpoints", "pods", ]
+    verbs      = ["get", "list", "watch", ]
+  }
+
+  rule {
+    api_groups = ["extensions", ]
+    resources  = ["ingresses", ]
+    verbs      = ["get", "list", "watch", ]
+  }
+
+  rule {
+    non_resource_urls = ["/metrics", ]
+    verbs             = ["get", ]
+  }
+}
+
+resource "kubernetes_cluster_role_binding" "prometheus" {
+  count = var.enable_monitoring ? 1 : 0
+
+  metadata {
+    name = "prometheus"
+  }
+
+  role_ref {
+    api_group = "rbac.authorization.k8s.io"
+    kind      = "ClusterRole"
+    name      = "prometheus"
+  }
+
+  subject {
+    kind      = "ServiceAccount"
+    name      = "default"
+    namespace = "monitoring"
+  }
+
+}
+
+resource "kubernetes_config_map" "prometheus" {
+  count = var.enable_monitoring ? 1 : 0
+
+  metadata {
+    name      = "prometheus-server-conf"
+    namespace = "monitoring"
+  }
+
+  data = {
+    "prometheus.rules" = "${file("${path.module}/config/prometheus/${var.config}.prometheus.rules")}"
+    "prometheus.yml"   = "${file("${path.module}/config/prometheus/${var.config}.prometheus.yml")}"
+  }
+
+}
+
+resource "kubernetes_deployment" "prometheus" {
+  count = var.enable_monitoring ? 1 : 0
+
+  metadata {
+    name      = "prometheus"
+    namespace = "monitoring"
+  }
+
+  spec {
+    replicas = 1
+
+    selector {
+      match_labels = {
+        app = "prometheus-server"
+      }
+    }
+
+    template {
+      metadata {
+        labels = {
+          app = "prometheus-server"
+        }
+      }
+
+      spec {
+        container {
+          image = "prom/prometheus:${var.prometheus_version}"
+          name  = "prometheus"
+
+          args = [
+            "--storage.tsdb.retention.time=12h",
+            "--config.file=/etc/prometheus/prometheus.yml",
+            "--storage.tsdb.path=/prometheus/",
+          ]
+
+          port {
+            container_port = 9090
+          }
+
+          resources {
+            limits = {
+              cpu    = 1
+              memory = "1Gi"
+            }
+            requests = {
+              cpu    = "500m"
+              memory = "500M"
+            }
+          }
+
+          volume_mount {
+            mount_path = "/etc/prometheus/"
+            name       = "prometheus-config-volume"
+          }
+
+          volume_mount {
+            mount_path = "/prometheus/"
+            name       = "prometheus-storage-volume"
+          }
+        }
+
+        volume {
+          name = "prometheus-config-volume"
+          config_map {
+            #              default_mode = "0420"
+            name = "prometheus-server-conf"
```

Comment:
Can you add dependency to the configmap explicitly? Probably kubernetes_config_map.prometheus.metadata.name

Reply:
done